frontend/src/components/Dashboard/page.tsx

```diff
@@ -14,45 +13,28 @@ const Dashboard = () => {
   const [selectedLocationPop, setSelectedLocationPop] = useState(null);
   const [expandedLocations, setExpandedLocations] = useState({});
   const [showPopup, setShowPopup] = useState(false);
-  const [memos, setMemos] = useState({});
-  const [reloadDashboard, setReloadDashboard] = useState(true);
+  const [memos, setMemos] = useState<any[]>([]);
   const [popReload, setPopReload] = useState(true);
   const { currentUser } = useFirebaseAuth();
 
-  // FIXME -- see line 100; handleDeleteMemo()
-  const [userData, setUserData] = useState({
-    memo: [],
-  });
-
   const fetchData = async () => {
     try {
-      const data = await UserController.get_user_data(currentUser.uid);
-      setUserData(data);
-      fetchMemos(data.memos);
-    } catch (error) {
-      console.error("Error fetching user data:", error);
-    }
-  };
-
-  const fetchMemos = async (memoIDs: string[]) => {
-    try {
-      const fetchedMemos: { [key: string]: MemoType } = {};
+      const memoData = await MemoController.get_all_memos(currentUser.uid);
+      setMemos(memoData);
       const fetchedLocations: { [key: string]: [number, number] } = {};
-
-      for (const mid of memoIDs) {
-        const data = await MemoController.get_memo(currentUser.uid, mid);
-        fetchedMemos[mid] = data;
-        fetchedLocations[data.location.name] = data.location.coordinates;
-      }
-
-      setMemos((prevMemos) => ({ ...prevMemos, ...fetchedMemos }));
+      memoData.forEach((memo) => {
+        fetchedLocations[memo.location.name] = memo.location.coordinates;
+      });
+  
       setLocations((prevLocations) => ({
         ...prevLocations,
         ...fetchedLocations,
       }));
+
     } catch (error) {
-      console.error("Error fetching memos:", error);
+      console.error("Error fetching data:", error);
```

Comment:
properly catch the errors here. This is for things like 5XX, etc. 

If the error was a 404 it means that there isn't any error, just that the user doesn't have any memos so you can silently "fail".

Reply:
so something like 
catch (error) {
      if (error.response && error.response.status >= 500 && error.response.status < 600) {
        console.error("Server Error:", error);
      } 
    }
and ignore 404?

Reply:
just have a case for 404 to return